### PR TITLE
Add unit test summary jobs to sharktank and shortfin workflows.

### PR DESCRIPTION
**Pull request by ScottTodd**
Progress on https://github.com/nod-ai/shark-ai/issues/357.

This gives us a single job in each unit test workflow to set as a "required check" that will block merging, replacing the current single `Unit Tests (3.11, 2.3.0, ubuntu-24.04)` check: 
image

I'm also switching ci-libshortfin.yml to run on all changes, which is a requirement for making a check required. We could still conditionally skip jobs based on paths modified... just not using GitHub's built in path filtering.

### CODE REVIEW

File: .github/workflows/ci-libshortfin.yml [hunk 1]
```diff
@@ -152,3 +146,23 @@ jobs:
...
+          echo '${{ toJson(needs) }}'
+          FAILED_JOBS="$(echo '${{ toJson(needs) }}' \
+            | jq --raw-output \
+            'map_values(select(.result!="success" and .result!="skipped")) | keys | join(",")' \
+          )"
+          echo "failed-jobs=${FAILED_JOBS}" >> $GITHUB_OUTPUT
+          if [[ "${FAILED_JOBS}" != "" ]]; then
+            echo "The following jobs failed: ${FAILED_JOBS}"
+            exit 1
+          fi
```
Comment (ScottTodd) [inline .github/workflows/ci-libshortfin.yml:168]: This failed jobs list code is forked from IREE. We could simplify for cases where there is only one job or we aren't using the `GITHUB_OUTPUT` (e.g. to send an alert to an external service).

Comment (marbre) [reply .github/workflows/ci-libshortfin.yml:168]: I don't know if there are plans to send it out via `GITHUB_OUTPUT` thus simplifying sounds good to me and if really needed we know where to fork the complexity from :smile:

Comment (ScottTodd) [reply .github/workflows/ci-libshortfin.yml:168]: Okay, I'll simplify. Will need to test a bit :P

Comment (ScottTodd) [reply .github/workflows/ci-libshortfin.yml:168]: Could just `echo "The job matrix failed"` I guess...

Comment (ScottTodd) [reply .github/workflows/ci-libshortfin.yml:168]: Eh. Supporting a list of jobs is still useful for the future. I'll just remove the currently unused `echo "failed-jobs=${FAILED_JOBS}" >> $GITHUB_OUTPUT`.

**Review by marbre — APPROVED** [review]

**Pull request merged**